### ISSUE TITLE
Error: Could not prefetch mailalias provider 'augeas': Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0)

### ISSUE THREAD

**Issue by vekexasia** [hercules-team/augeasproviders#141]
Hello i've a simple mailalias directive that throws this error :(

the full error is 
```
Error: Could not prefetch mailalias provider 'augeas': Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0)
Error: Could not set 'present' on ensure: Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0) at 122:/etc/puppet/environments/production/manifests/monba.pp
Error: Could not set 'present' on ensure: Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0) at 122:/etc/puppet/environments/production/manifests/monba.pp
Wrapped exception:
Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0)
Error: /Stage[main]/Main/Node[monba]/Mailalias[root]/ensure: change from absent to present failed: Could not set 'present' on ensure: Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0) at 122:/etc/puppet/environments/production/manifests/monba.pp
```
the code is

```
mailalias { 'root':
		ensure    => 'present',
		recipient => '*****@gmail.com';
	}
```

Can you tell me whats' wrong ?

**Comment by vekexasia** [comment]
/var/lib/puppet/lib/augeas/lenses contains only "known_hosts.aug"

**Comment by domcleal** [comment]
> Augeas didn't load /etc/aliases with Aliases.lns from /var/lib/puppet/lib/augeas/lenses: Iterated lens matched less than it should (line:6, character:0)

There's something in /etc/aliases on line 6 that the Augeas lens (provided by the package, not /var/lib/puppet) is unable to parse.  Check the contents of line 6, it might be a bug in Augeas that's fixed in a later version.

**Comment by vekexasia** [comment]
Indeed, I truncated /etc/aliases and re-run puppet now it works. The file was almost identical to what it is now. I'm a bit confused ....

**Comment by raphink** [comment]
@vekexasia Can you post the `/etc/aliases` lines that fail?

**Comment by vekexasia** [comment]
Unfortunately I already truncated the file :( i'm unsure how I might restore it. I believe, i had:
```
root:***@gmail.com
```
without the space.

**Comment by raphink** [comment]
@vekexasia The lens knows how to parse that properly. So I'll close the bug then, if there's no trace of the line you truncated.

**Comment by vekexasia** [comment]
I confirm that if i've "root:email@gmail.com" without the space i get that error.

**Comment by domcleal** [comment]
https://github.com/hercules-team/augeas/commit/011a0ec09e4ed8c2f7cd848eaf799eee0838c356 - fixed in Augeas 1.3.0.
